lib/fluent/plugin/base.rb

```diff
@@ -51,6 +51,10 @@ def fluentd_worker_id
         @_fluentd_worker_id
       end
 
+      def fluentd_lockdir
+        @_fluentd_lockdir ||= ENV['FLUENTD_LOCKDIR']
```

Comment:
It should be initialized at `initialize`.

Reply:
Addeed initialization (in the same way with fluentd_worker_id).

Reply:
Why don't you put `@fluentd_lockdir = ENV['FLUENTD_LOCKDIR']`, remove this method and use `@fluentd_lockdir` directly?

BTW, `lock_dir` may be better than `lockdir` because `plugin_root_dir` uses `_` before `dir`.

Reply:
Hmm, `@_fluentd_worker_id` is designed that way.
Does this mean that the environment variable can be set afterwards?

Reply:
> Why don't you put @fluentd_lockdir = ENV['FLUENTD_LOCKDIR'], remove this method and use @fluentd_lockdir directly?

I fixed the initialization part following this.